pennylane/labs/tests/resource_estimation/test_resource_tracking.py

```diff
@@ -0,0 +1,355 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Test the core resource tracking pipeline."""
+from collections import defaultdict
+from copy import copy
+
+# pylint:disable=protected-access, no-self-use
+import pytest
+
+import pennylane as qml
+import pennylane.labs.resource_estimation as re
+from pennylane.labs.resource_estimation.resource_tracking import (
+    DefaultGateSet,
+    _clean_gate_counts,
+    _counts_from_compressed_res_op,
+    _operations_to_compressed_reps,
+    _StandardGateSet,
+    get_resources,
+    resource_config,
+    resources_from_operation,
+    resources_from_qfunc,
+    resources_from_tape,
+)
+
+
+class DummyOperation(qml.operation.Operation):
+    """Dummy class to test _operations_to_compressed_reps function."""
+
+    def __init__(self, wires=None):
+        super().__init__(wires=wires)
+
+    def decomposition(self):
+        decomp = [
+            re.ResourceHadamard(wires=self.wires[0]),
+            re.ResourceHadamard(wires=self.wires[1]),
+            re.ResourceCNOT(wires=[self.wires[0], self.wires[1]]),
+        ]
+
+        return decomp
```

Comment:
Why is this necessary? Are the resources classes we already have implemented not sufficient?

Reply:
This is the base PR that the other PRs pull from so we don't have access to those objects here. Also in general unit testing base classes is done using dummy operations to reduce the coupling between changes in other operations causing tests to fail for other objects.